core/src/main/java/org/infinispan/persistence/sifs/Compactor.java

```diff
@@ -668,15 +670,17 @@ private void compactSingleFile(int scheduledFile, boolean isLogFile, CompactionE
             // compaction before the index has been fully updated. Thus we block any other compaction events
             // until all entries have been moved for this file
             CompletionStage<Void> aggregate = aggregateCompletionStage.freeze();
-            paused = new CompletableFuture<>();
-            // We resume after completed, Note that we must complete the {@code paused} variable inside the compactor
-            // execution pipeline otherwise we can invoke compactor operations in the wrong thread
-            aggregate.whenComplete((ignore, t) -> {
-               resumeAfterPause();
-               if (t != null) {
-                  log.error("There was a problem moving indexes for compactor with file " + logFile.fileId, t);
-               }
-            });
+            if (!CompletionStages.isCompletedSuccessfully(aggregate)) {
```

Comment:
This is an unrelated change to both issues, but I am sneaking it in here. It allows for compactor to run a tad more efficiently when all the index updates were already completed. This avoids additional context switches.